### PR TITLE
refactor(cache/package): Enable strict checks

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Convert functions to `async/await` if needed
- Clarify return types
- Additional guards

## Context:

- Ref: #7154
- Blocked by: #12842

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/cache/package/index.ts [hunk 1]
```diff
@@ -10,29 +10,33 @@ function getGlobalKey(namespace: string, key: string): string {
...
   minutes: number
 ): Promise<void> {
   if (!cacheProxy) {
-    return undefined;
+    return;
   }
   const globalKey = getGlobalKey(namespace, key);
   memCache.set(globalKey, value);
-  return cacheProxy.set(namespace, key, value, minutes);
+  await cacheProxy.set(namespace, key, value, minutes);
```
Comment (rarkins) [inline lib/util/cache/package/index.ts:39]: Was this a coding mistake before now?

File: lib/util/cache/package/redis.ts [hunk 2]
```diff
@@ -25,14 +25,14 @@ async function rm(namespace: string, key: string): Promise<void> {
...
-): Promise<T> {
+): Promise<T | undefined> {
   if (!client) {
     return undefined;
   }
   logger.trace(`cache.get(${namespace}, ${key})`);
   try {
     const res = await client?.get(getKey(namespace, key));
-    const cachedValue = JSON.parse(res);
+    const cachedValue = res && JSON.parse(res);
```
Comment (rarkins) [inline lib/util/cache/package/redis.ts:35]: Another possible coding mistake?

**Review by rarkins — APPROVED** [review]

File: lib/util/cache/package/index.ts (hunk 1 above)
Comment (zharinov) [reply lib/util/cache/package/index.ts:39]: Not exactly, it were more liberal `async/await` usage rules

Comment (viceice) [reply lib/util/cache/package/index.ts:39]: Seems so

File: lib/util/cache/package/redis.ts (hunk 2 above)
Comment (viceice) [reply lib/util/cache/package/redis.ts:35]: Yes, can throw

**Review by viceice — APPROVED** [review]

**Pull request merged**